### PR TITLE
Fixing multi-release *-source.jar creation - always put versioned sources to META-INF/versions/ jar subfolder

### PR DESCRIPTION
**Pull request by ropalka**

### CODE REVIEW

**Comment by ropalka** [comment]
@dmlloyd please review

File: pom.xml [hunk 1]
```diff
@@ -1097,6 +1144,45 @@
...
+                        <artifactId>maven-resources-plugin</artifactId>
+                        <executions>
+                            <execution>
+                            <id>copy-multirelease-sources-java10</id>
+                            <phase>prepare-package</phase>
+                            <goals>
+                                <goal>copy-resources</goal>
+                            </goals>
+                            <configuration>
+                            <outputDirectory>${java10.sources.metainf.location}/META-INF/versions/10</outputDirectory>
```
Comment (dmlloyd) [inline pom.xml:1158]: Why can the sources not be included directly from their original locations?

Comment (ropalka) [reply pom.xml:1158]: I gave it more try and the answer is yes, it is possible to copy them directly @dmlloyd .

**Review by jamezp — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -1053,6 +1053,27 @@
...
+                                <id>include-multirelease-sources-java9</id>
+                                <phase>generate-resources</phase>
+                                <goals>
+                                    <goal>add-resource</goal>
+                                </goals>
+                                <configuration>
+                                    <resources>
+                                        <resource>
+                                            <directory>${project.basedir}/src/main/java9</directory>
+                                            <targetPath>META-INF/versions/9</targetPath>
```
Comment (dmlloyd) [inline pom.xml:1070]: Can you confirm that this would only add the sources to the source JAR and not the main artifact?

Comment (ropalka) [reply pom.xml:1070]: Good catch @dmlloyd ! Of course you are right, with this configuration also MR jar file contained the java source files which is something we don't want.

**Comment by ropalka** [comment]
Superseded by https://github.com/jboss/jboss-parent-pom/pull/143

**Pull request closed without merging**